packages/visual-editor/src/internal/components/InternalLayoutEditor.tsx

```diff
@@ -169,11 +173,11 @@ export const InternalLayoutEditor = ({
         alert("Failed to paste: Invalid component data.");
         return;
       }
-      // Update the component data in the specified zone
+
       dispatch({
-        type: "set",
-        state: (prevState) => {
-          const newContent = [...prevState.data.content];
+        type: "setData",
```

Comment:
There is a "replace" type now. Does that make more sense? You'll still need to preserve the id of the target component.